Source/DafnyLanguageServer/Language/Symbols/DafnyLangSymbolResolver.cs

```diff
@@ -150,9 +154,43 @@ private class SymbolDeclarationResolver {
           cancellationToken.ThrowIfCancellationRequested();
           functionSymbol.Parameters.Add(ProcessFormal(scope, parameter));
         }
+        if (function.Result != null) {
+          functionSymbol.Parameters.Add(ProcessFormal(scope, function.Result));
+        }
+        ScopeSymbol? ExpressionHandler(Expression? expression) =>
+          expression == null
+            ? null
+            : ProcessExpression(new ScopeSymbol(functionSymbol, functionSymbol.Declaration), expression);
+        functionSymbol.Body = ExpressionHandler(function.Body);
+        functionSymbol.Ensures.AddRange(ProcessListAttributedExpressions(function.Ens, ExpressionHandler));
+        functionSymbol.Requires.AddRange(ProcessListAttributedExpressions(function.Req, ExpressionHandler));
+        functionSymbol.Reads.AddRange(ProcessListFramedExpressions(function.Reads, ExpressionHandler));
+        functionSymbol.Decreases.AddRange(ProcessListExpressions(function.Decreases.Expressions, ExpressionHandler));
+        functionSymbol.ByMethodBody = function.ByMethodBody == null ? null : ProcessFunctionByMethodBody(functionSymbol, function.ByMethodBody);
         return functionSymbol;
       }
 
+      private IEnumerable<ScopeSymbol> ProcessListAttributedExpressions(
+        IEnumerable<AttributedExpression> list, Func<Expression?, ScopeSymbol?> expressionHandler
+        ) {
+        return list.Select(attributedExpression =>
+          expressionHandler(attributedExpression.E)).Where(x => x != null).OfType<ScopeSymbol>();
```

Comment:
Sorry, that's probably a bit nitpicking. Prefer using `Cast<ScopeSymbol>` here, for the same reason to prefer c-style casts over `as` casts.

Would you mind changing the format to something like this? You can leave that, it's just a note for consistency.
```cs
        return list.Select(attributedExpression =>expressionHandler(attributedExpression.E))
          .Where(x => x != null)
          .OfType<ScopeSymbol>();
```